hypershift-operator/controllers/hostedcluster/internal/platform/ibmcloud/ibmcloud_test.go

```diff
@@ -0,0 +1,147 @@
+			expectedObject:   nil,
+		},
+		"when VPC provider type specified for IBMCloud no CAPI cluster is created": {
+			inputControlPlaneNamespace: fakeControlPlaneNamespace,
+			inputHostedCluster: &hyperv1.HostedCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fakeHostedClusterName,
+					Namespace: fakeHostedClusterNamespace,
+				},
+				Spec: hyperv1.HostedClusterSpec{
+					Platform: hyperv1.PlatformSpec{
+						Type: hyperv1.IBMCloudPlatform,
+						IBMCloud: &hyperv1.IBMCloudPlatformSpec{
+							ProviderType: v1.IBMCloudProviderTypeVPC,
+						},
+					},
+				},
+			},
+			inputAPIEndpoint: fakeAPIEndpoint,
+			expectedObject:   nil,
+		},
+		"when UPI provider type specified for IBMCloud no CAPI cluster is created": {
+			inputControlPlaneNamespace: fakeControlPlaneNamespace,
+			inputHostedCluster: &hyperv1.HostedCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fakeHostedClusterName,
+					Namespace: fakeHostedClusterNamespace,
+				},
+				Spec: hyperv1.HostedClusterSpec{
+					Platform: hyperv1.PlatformSpec{
+						Type: hyperv1.IBMCloudPlatform,
+						IBMCloud: &hyperv1.IBMCloudPlatformSpec{
+							ProviderType: v1.IBMCloudProviderTypeVPC,
+						},
+					},
+				},
+			},
+			inputAPIEndpoint: fakeAPIEndpoint,
+			expectedObject:   nil,
+		},
+		"when platform type not specified for IBMCloud a VPC CAPI cluster is created": {
+			inputControlPlaneNamespace: fakeControlPlaneNamespace,
+			inputHostedCluster: &hyperv1.HostedCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fakeHostedClusterName,
+					Namespace: fakeHostedClusterNamespace,
+				},
+				Spec: hyperv1.HostedClusterSpec{
+					Platform: hyperv1.PlatformSpec{
+						Type:     hyperv1.IBMCloudPlatform,
+						IBMCloud: &hyperv1.IBMCloudPlatformSpec{},
+					},
+				},
+			},
+			inputAPIEndpoint: fakeAPIEndpoint,
+			expectedObject: &capiibmv1.IBMVPCCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: fakeControlPlaneNamespace,
+					Name:      fakeHostedClusterName,
+					Annotations: map[string]string{
+						capiv1.ManagedByAnnotation: "external",
+					},
+					// since resource created through fakeclient this is set to 1 to ensure the struct compare works
+					ResourceVersion: "1",
+				},
+				Status: capiibmv1.IBMVPCClusterStatus{
+					Ready: true,
+				},
+				Spec: capiibmv1.IBMVPCClusterSpec{
+					ControlPlaneEndpoint: capiv1.APIEndpoint{
+						Port: fakeAPIEndpoint.Port,
+						Host: fakeAPIEndpoint.Host,
+					},
+				},
+			},
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			g := NewGomegaWithT(t)
+			fakeUpdater := &simpleCreateOrUpdater{}
+			fakeClient := fake.NewClientBuilder().WithScheme(api.Scheme).Build()
+			fakeReconciler := IBMCloud{}
+			actualinfraCR, err := fakeReconciler.ReconcileCAPIInfraCR(context.Background(), fakeClient, fakeUpdater.CreateOrUpdate, test.inputHostedCluster, test.inputControlPlaneNamespace, test.inputAPIEndpoint)
```

Comment:
There is no need for the `simpleCreateOrUpdater` at all:
```suggestion
			actualinfraCR, err := fakeReconciler.ReconcileCAPIInfraCR(context.Background(), fakeClient, controllerutil.CreateOrUpdate, test.inputHostedCluster, test.inputControlPlaneNamespace, test.inputAPIEndpoint)
```